lib/improver/utilities/ancillary_creation.py

```diff
@@ -0,0 +1,154 @@
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Plugins to create ancillary data."""
+
+import iris
+import numpy as np
+import improver.constants as cc
+from improver.psychrometric_calculations.psychrometric_calculations import (
+    WetBulbTemperature)
+
+
+class SaturatedVapourPressureTable(object):
+
+    """
+    Plugin to create a saturated vapour pressure lookup table.
+    """
+
+    def __init__(self, T_min=183.15, T_max=338.15, T_increment=0.1):
+        """
+        Initialise class.
+
+        Args:
+            T_min : float
+                The minimum temperature for the range.
+            T_max : float
+                The maximum temperature for the range.
+            T_increment : float
+                The temperature increment at which to create values for the
+                saturated vapour pressure between T_min and T_max.
+        """
+        self.T_min = T_min
+        self.T_max = T_max
+        self.T_increment = T_increment
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<SaturatedVapourPressureTable: T_min: {}; T_max: {}; '
+                  'T_increment: {}>'.format(self.T_min, self.T_max,
+                                            self.T_increment))
+        return result
+
+    def saturation_vapour_pressure_goff_gratch(self, temperature):
+        """
```

Comment:
It would make more sense to have this calculation with the other calculations in the Utilities class of psychrometric_calculation.py but you can leave this here if you want.

Reply:
Agreed. I've moved it.

Reply:
In doing this I also changed the calculation of the temperatures for the lookup table from using np.linspace to use np.arange. I did this as linspace does not ensure consistent increments, though the deviation was small. As a result I have also recreated the svp_table.py data so that the default output of this ancillary creation routine matches. I've checked the output against the UM once again. The maximum difference remains unchanged (0.0055K) and the average deviation reduces to 0.00024K.